app/models/cart.rb

```diff
@@ -10,7 +10,23 @@ def total
   end
 
   def add_item(id)
+    if @contents[id] == nil
```

Comment:
this nil condition should never happen because Hash.new(0) is going to set every key as 0